peer/x/roundrobin/list.go

```diff
@@ -318,3 +320,46 @@ func (pl *List) handleUnavailablePeerStatusChange(p peer.Peer) error {
 
 	return pl.addToAvailablePeers(p)
 }
+
+// Introspect returns a ChooserStatus with a summary of the Peers.
+func (pl *List) Introspect() introspection.ChooserStatus {
+	state := "Stopped"
+	if pl.IsRunning() {
+		state = "Running"
+	}
+
+	pl.lock.Lock()
+	availables := pl.availablePeerRing.All()
+	unavailables := make([]peer.Peer, 0, len(pl.unavailablePeers))
+	for _, peer := range pl.unavailablePeers {
+		unavailables = append(unavailables, peer)
+	}
+	pl.lock.Unlock()
+
+	peersStatus := make([]introspection.PeerStatus, 0,
+		len(availables)+len(unavailables))
+
+	buildPeerStatus := func(peer peer.Peer) introspection.PeerStatus {
+		return introspection.PeerStatus{
+			Identifier: peer.Identifier(),
+			State: fmt.Sprintf("%s, %d pending request(s)",
+				peer.Status().ConnectionStatus.String(),
+				peer.Status().PendingRequestCount),
```

Comment:
Presumably `peer.Status()` requires some lock or atomic operation - can we instead call it once and re-use the returned value when building the state?